flink-tests/src/test/java/org/apache/flink/test/util/TestUtils.java

```diff
@@ -84,4 +92,32 @@ public static void waitUntilJobInitializationFinished(
                 () -> clusterClient.requestJobResult(id).get(),
                 userCodeClassloader);
     }
+
+    public static File getMostRecentCompletedCheckpoint(File checkpointDir) throws IOException {
+        return Files.find(checkpointDir.toPath(), 2, TestUtils::isCompletedCheckpoint)
```

Comment:
FYI there is a bug here https://issues.apache.org/jira/browse/FLINK-23647. But we would fix it with that ticket.

Reply:
Which ticket do you mean by `that` ticket?